### DISCUSSION TITLE
Q4S1 (Oct to Mid-Nov) Roadmap Planning

### DISCUSSION THREAD

**Discussion by BenJam** [opencollective/opencollective#4746] [Ideas]
![yourarehere](https://user-images.githubusercontent.com/158833/135057586-76173453-a4c4-4276-aa66-2ec13f8187c3.png)

The leaves are turning, the temperatures are dropping and _everyone in NZ is getting ready for summer!_ but regardless of the changing of the seasons our focus does not change... 

[Last sprint](https://github.com/opencollective/opencollective/issues/4711) we continued focussing on consolidating on what we have already designed and developed, including launching some features that were ready to go (hello crpyto and stock contributions!). We also readied some design work to improve our funnel processes for Collectives, Hosts and to tell the story of Open Collective more clearly and distinct from the work of the hosts we operate. 

## Our priorities

> Our priorities do not change, but our framing does. We continue to focus on our core strengths but as a product team, we shall consolidate on our existing success.

In order to free the team to develop first and third-party hosts, and to focus on the things that are often lost in the heat of feature development, we felt that consolidating our success should be the focus for the remainder of the year. That we should focus on fundamentals like the ledger, completing unfinished projects and, addressing issues that cause ourselves and our user's significant pains and limit our growth.

So what do we _not_ prioritize? We should try not to focus on the needs of smaller hosts, self-hosted collectives, and significant, new feature development. This does not mean that we ignore these needs, and we should continue to listen to our users and understand their concerns, but we should seek to understand their impact on the whole rather than the select few. 

## How do I contribute?

This is the _fourth_ time we've used GitHub Discussions to collect ideas and vote on priorities, so members of the core team should be used to the mechanics (described below). For our hosts who wish to contribute to the dicussion:  

#### Proposing issues
Create a top-level comment on this discussion, link to any issues that already exist in this repo (a quick search is good enough). Bear in mind issues that were discussed in previous sprints that we may have [added to the roadmap](https://github.com/opencollective/opencollective/projects/54). If you can't find an existing issue create one (if it's a dupe we'll merge them later) using one of the [user need](https://github.com/opencollective/opencollective/issues/new?assignees=&labels=&template=user_need.md&title=) (if you aren't proposing a solution) or [feature spec](https://github.com/opencollective/opencollective/issues/new?assignees=&labels=feature&template=feature-spec.md&title=) (if you have something in mind) templates.  

#### Triaging issues
Last sprint there were a large number of small issues proposed. It was difficult to discern which issues were a higher priority than others. As a result, we (okay @benjam) will be applying the following triage heuristic which should be applied to issues proposed for a sprint. There's [an explainer on Loom](https://www.loom.com/share/369ab467fbc64dec848085d38ff57ca0) if you'd like to know more. 

If you'd like to apply this heuristic to issues I recommend [adding the following as a saved reply in your GItHub settings](https://docs.github.com/en/github/writing-on-github/working-with-saved-replies/creating-a-saved-reply): 

<details>
<summary>Triage Template</summary>
Problem, user need or idea:
Proposed solution:

P1 high frequency high impact
P2 low frequency, high impact
P3 high frequency, low impact
P4 low frequency, low impact

Examples of high impact - a problem affects users during an essential process (expenses and payments > onboarding and registration > contributing) with no workaround.

High frequency - >10% of users affected (measured as a proportion of total potential users for this case). 
</details>

#### Scoping the problem
If you feel that the idea for a project or a specific issue isn't well-scoped then feel free to loop in others to help scope it down. This might involved validating the problem with some quantitative research, asking someone whether they've encountered anyone else with a similar problem or reaching out to users specifically to check an idea or validate an assumption. There is no need to scope the solution at this stage, we are looking for the most impactful problems to solve that are in line with our mission, our values, and our current focus. 

#### Discussion
Feel free to lend thoughts, opinions, data and ask questions in threads from each of the top-level comments (which should contain the proposed project).

#### Voting
Feel free to +1 any project ideas by hitting the ⬆️  button alongside any top-level comment. Try to limit yourself to three votes for the sprint to keep things focussed.

#### Synchronous discussion
We'll arrange a synchronous discussion for everyone **Tuesday 5th October** to talk through the top pitches for the quarter and to create the space for us to decide together based on the outcome of this discussion.

#### Kickoffs and further discussion
Depending on the level of shared understanding between product, design, and engineering we may not need a kick-off for each individual issue that that issue contains the necessary detail. A quick 1:1 1:many discussion to scope the issue(s) within a theme may get us through.

**Comment by znarf** [comment]
### Financial Contributions across Hosts

https://github.com/opencollective/opencollective/issues/3190

**Reply by alanna** [reply]
We decided in a team meeting that expenses across hosts was the priority over contributions

**Reply by znarf** [reply]
Agree.

**Comment by znarf** [comment]
### Expenses across Hosts

This is a requirement for Google Season of Docs and should have a relatively low complexity if based on simple Bank Transfers between Hosts.

This can also serve as a basis for for an Expense based for Contribution across Hosts.

https://github.com/opencollective/opencollective/issues/4497 is also somehow related to this.

**Comment by znarf** [comment]
### Virtual Cards with Stripe

**Reply by alanna** [reply]
In specific, we need to confirm a) what exactly the cause of the issues with Privacy are (I mean about pausing the account - the US limitation is clear), and b) if Stripe will solve them.

**Reply by BenJam** [reply]
++ I have made an enquiry with Stripe and I have a conversation with them this evening or Wednesday evening to find out more.

**Reply by BenJam** [reply]
okay so we already signed up for issuing with stripe, I spoke with one of their staff last night and they 👍  to support us. And it looks like they will take us on a basic, commercial plan. Some details that I can share but given the issues we're having on the Open Collective Foundation side at the moment it's worth prioritising the move.

**Comment by Betree** [comment]
### Legal name followup

Followup on https://github.com/opencollective/opencollective/issues/3689. The designs for the contribution flow were not ready when we began the sprint, so we still need to introduce the field to the contribution flow.

Overall, this is a small project that shouldn't take long.

Issues to tackle:
- https://github.com/opencollective/opencollective/issues/4692
- https://github.com/opencollective/opencollective/issues/4654
- https://github.com/opencollective/opencollective/issues/4719 
- https://github.com/opencollective/opencollective/issues/4752

**Reply by alanna** [reply]
I think it's important we complete projects already started and close to being finished and not leave loose ends.

**Reply by BenJam** [reply]
++

**Comment by Betree** [comment]
### Admin dashboard reports

Followup on https://github.com/opencollective/opencollective/issues/4300. We still have some work on that part, mostly timeseries and updates to the existing page to make it look better. This project sounds like a priority, as the page looks incomplete in its current version.

Issues that I hope will be finished and merged this week:
- https://github.com/opencollective/opencollective/issues/4658
- https://github.com/opencollective/opencollective/issues/4660

Issues that I imagine will be part of next sprint:
- https://github.com/opencollective/opencollective/issues/4726
- https://github.com/opencollective/opencollective/issues/4740
- https://github.com/opencollective/opencollective/issues/4690
- https://github.com/opencollective/opencollective/issues/4606 (historical view)
- https://github.com/opencollective/opencollective/issues/4657 (historical view)

**Comment by piamancini** [comment]
We have some spillover from this Q (admin dashboard, legal names, likely crypto). These issues + Stripe for virtual cards which is high impact and high frequency seem like a good number of projects.

**Comment by alanna** [comment]
# OCF Priorities

The OCF team has done a prioritisation process and this is the outcome.

## Quick Wins

*  [Invited expenses UX quick fix](https://github.com/opencollective/opencollective/issues/4751#issuecomment-932678959) ✅ 
*  [Show expense payee total this year](https://github.com/opencollective/opencollective/issues/4491)✅
*  [No tax form triggered for Grant expenses](https://github.com/opencollective/opencollective/issues/4766)✅ 
*  [Refund added funds transactions](https://github.com/opencollective/opencollective/issues/4369)❌
*  [Blog and docs retain nav bar](https://github.com/opencollective/opencollective/issues/4761)❌
*  [Host can automatically tweet when a new Collective is approved](https://github.com/opencollective/opencollective/issues/4765)❌
*  [Host can set custom manual payment methods](https://github.com/opencollective/opencollective/issues/4024)❌

## Marketing & Comms

*  [More social sharing buttons](https://github.com/opencollective/opencollective/issues/4762)✅ 
*  [Embed video in main Collective page as its own section](https://github.com/opencollective/opencollective/issues/4763)✅ 
*  [Embed photo gallery in main Collective page as its own section](https://github.com/opencollective/opencollective/issues/4767)✅ 
*  Memo’s work on attractive social media linking (no issue created it seems, but @memo-es demoed it recently)✅ 
*  [Host changelog updates through fire](https://github.com/opencollective/opencollective/issues/4708)❌
*  [Search based on tags like categories, locations](https://github.com/opencollective/opencollective/issues/4759)❌
*  [Suggest (or force) tags during Collective creation](https://github.com/opencollective/opencollective/issues/4764)❌
*  [Fire for new users linking to an initial walkthrough](https://github.com/opencollective/opencollective/issues/4768)✅ 
*  [Reveal ‘contact’ action instead of hiding it under actions menu](https://github.com/opencollective/opencollective/issues/4760)✅

## Money Movements 

* [Invite org to create expense](https://github.com/opencollective/opencollective/issues/4224)❌
* [Payable invoice attached to pending order](https://github.com/opencollective/opencollective/issues/4497)❌
* [Invited expenses UX proper fix](https://github.com/opencollective/opencollective/issues/4751)✅
* [Multiple expense approvals](https://github.com/opencollective/opencollective/issues/3081)✅
* [Edit pending order](https://github.com/opencollective/opencollective/issues/4069)❌

## Needs design/scoping

Not yet clear enough to create issues for. Mentioning as suggestions for the design queue so we can potentially prioritise these in the future.

* Fix issues with Virtual cards getting shut down ✅
* Host discoverability ✅
* Improved ‘Discover’ experience ✅
* Improved access to help docs and improved help docs, e.g. a chat bot before you email support ✅
* Accessibility/inclusion improvements ✅
* Host can create fee waiver campaign ❌

**Reply by alanna** [reply]
The other issues I'm posting in this discussion are my own priorities with my general OC hat on, not representing OCF collectively.

**Reply by alanna** [reply]
For reference, about 40% of OC's income is from OCF so I guess we should weight things accordingly?

**Comment by alanna** [comment]
## #4754 Separate activation email for new Independent Collectives

I think this is an easy way to save time for @piamancini and support who end up holding these users' hands a lot. I want to link to the [new IC docs](https://docs.opencollective.com/help/independent-collectives) and an upcoming video demo for them.

**Comment by alanna** [comment]
## #4461 Make reply to email for team addition notification the adder only

I think we really should make small tweaks like this that actually could meaningfully reduce support load.

**Comment by alanna** [comment]
## #4056 Remove the Host Plan selection settings

Still leftover from the new pricing model project, and needlessly complex. Even more important now that we're trying to separate out Independent Collectives from Fiscal Hosts conceptually and simplify.

**Comment by alanna** [comment]
## #3865 Do not treat system-generated expense summary as an attachment

Small change that would really help host admins.

**Comment by alanna** [comment]
## #2808 Host admin can export Collective data CSV

This one got lots of +1s in the previous sprint planning discussion and I still think it's important.

**Comment by alanna** [comment]
## #3330 Host admin can unhost Collective

Legal/compliance ticking time bomb not to have this.

**Comment by alanna** [comment]
@BenJam I have gone through my requested issued in the previous discussion and added the ones I think should be considered again this round here (not counting the ones already covered by the OCF priorities post above), so you don't need to go back through them.

**Comment by SudharakaP** [comment]
## Improvements to Crypto Contributions

- https://github.com/opencollective/opencollective/issues/4649
- https://github.com/opencollective/opencollective/issues/4616

**Reply by BenJam** [reply]
Should we consider the testing a pickup from the previous sprint or is there enough coverage to consider this safe unless we're working around the feature? in my opinion we should lead on marketing and see what the usage is like before spending any more time on it.

**Reply by SudharakaP** [reply]
Makes sense; if we decide to scratch off Crypto contributions, time spent on this might not be worthwhile. So maybe give it a bit more time before contributing to this. But I am open to doing these this sprint as well. 👍🏽

**Comment by znarf** [comment]
## Transactions CSV follow up

After discussion with accountants.

- Surface Expense Tags
- Surface Merchant reference number (Stripe, PayPal, Wise)
- View and download "hostTransactions" from the interface
- Reference between Refunds and Refunded transactions

**Comment by Memo-Es** [comment]
## Marketing update
* About us page (Illustrations)
* Homepage (Continuation) https://github.com/opencollective/opencollective/issues/4689
* https://github.com/opencollective/opencollective/issues/4762
* [Marketing assets (Continuation)](https://www.loom.com/share/9a3fe92e77514f7db16320ea41b71c57).

**Comment by Memo-Es** [comment]
## Design small projects
* Add projects and events to host dashboard
* https://github.com/opencollective/opencollective/issues/4751#issuecomment-932678959
* **Collective page update**
     * https://github.com/opencollective/opencollective/issues/4760
     * https://github.com/opencollective/opencollective/issues/4767
     * https://github.com/opencollective/opencollective/issues/4763
* https://github.com/opencollective/opencollective/issues/4750
* https://github.com/opencollective/opencollective/issues/4768
* Transactions page design update

**Comment by Memo-Es** [comment]
## Visual documentation
* [Diagrams for docs (Continuation)](https://www.loom.com/share/e21457adca2d49fdb10823e741241fe1)
* Story board for OCF animation video w/Caroline

**Comment by Memo-Es** [comment]
## Design system update 
* Illustration system (Issue pending)
* Brand repository (OC, OSC, OCF)
* Enable design index

**Comment by BenJam** [comment]
## Add GDPR Notices to contribution flow

Implement the design in https://github.com/opencollective/opencollective/issues/4537#issuecomment-926489917 which makes it clearer what is shared and how to opt out.  Also remove contact email address from incognito contributions in csv export.

**Comment by BenJam** [comment]
## Two to sign & Multiple expense approvals

Two projects that were picked up last sprint that should be completed this. #3328 #3081 I need to compile them into a proper project. Note that #3081 is inc in @alanna's rundown.

**Comment by BenJam** [comment]
## Digital Wallet Payments #4327 

We implemented it, we were cut off by AliPay, we should just pick another provider (Apple, Google, Amazon or Payoneer). Features that we've implemented that aren't being used should be removed and I'd prefer not to remove this!

**Comment by BenJam** [comment]
## Projects Follow ups

**Comment by Betree** [comment]
# Security Hackathon

Maybe a bit late to suggest that, but with fresh eyes coming into the code I would love to organize a mini-hackathon about security. That could take the form of a single day that we dedicate to:
- Researching/pentesting: engineers can dedicate their day to testing issues, documenting problems. Having everyone doing this at the same time can create an emulation, sharing their findings and ideas.
- Fixing issues: we have some (mostly minor) security issues already documented in our backlog. 

**Participants**
- All engineers from the core team: It's a good practice to look at our code with critical eyes, and putting yourself in the shoes of an attacker from time to time later helps to write safer code.
- Existing contributors: That will be a great way to familiarize people with our security policy, and the process of reporting issues responsibly.

**Comment by SudharakaP** [comment]
## Create Issues or Use Existing Issues for [Hacktoberfest](https://hacktoberfest.digitalocean.com/)

- Add the `Hacktoberfest` tag to the OpenCollective repos; mainly `opencollective-frontend`, `opencollective-api`, `opencollective-rest`, `opencollective-pdf`, `documentation`, `opencollective-changelog`

- Maybe we need to select some smaller issues and label them with Hacktoberfest labels so that our repos participate in the Hacktoberfest; helping the community as well as getting some smaller paper-cut type issues to get solved. 🤔 😄 

Reference: https://hacktoberfest.digitalocean.com/resources/maintainers

**Reply by Betree** [reply]
> Maybe we need to select some smaller issues and label them with Hacktoberfest labels so that our repos participate in the Hacktoberfest; helping the community as well as getting some smaller paper-cut type issues to get solved. thinking smile

Isn't the `good first issue` enough?

**Reply by SudharakaP** [reply]
My thinking is adding the `Hacktoberfest` is better since most people will probably specifically search GitHub for this label. 🤔 Easy thing to do is just add `Hacktoberfest` to all `Good first Issue` issues maybe. 😉

**Reply by SudharakaP** [reply]
Also I noticed that we have a very few `Good first Issues`; https://github.com/opencollective/opencollective/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22. And some might not be very good for first time contributors. So we might need to create some more issues or add `Good first Issues/Hacktoberfest` tags to existing smaller issues if we decide to do this.

**Reply by Betree** [reply]
Yep definitely, `good first issue` should ideally be added on all simple/minimal issues that are simple to reproduce

**Reply by SudharakaP** [reply]
@Betree : Okay, thanks. So I'll make sure to add the `good first issue` label. As for hacktoberfest label I've added it for the `frontend` and `api` repos.
